bin/migration/tables/cases.py

```diff
@@ -0,0 +1,103 @@
+from .helpers import check_existing_record, parse_to_timestamp, audit_entry_creation, log_failed_imports
+import uuid
+
+class CaseManager:
+    def __init__(self, source_cursor):
+        self.source_cursor = source_cursor
+        self.failed_imports = set()
+
+    def get_data(self):
+        self.source_cursor.execute("SELECT * FROM public.cases")
+        return self.source_cursor.fetchall()
+
+    def migrate_data(self, destination_cursor, source_data):
+        # creating a temporary table for the unique case_ref values
+        destination_cursor.execute(
+            """CREATE TABLE IF NOT EXISTS public.temp_cases (
+                                    booking_id UUID PRIMARY KEY,
+                                    case_id UUID,
+                                    reference VARCHAR(25),
+                                    court_name VARCHAR(250),
+                                    court_id UUID,
+                                    created_at VARCHAR(50),
+                                    created_by VARCHAR(100),
+                                    modified_at VARCHAR(50)
+                                    )
+                                """
+        )
+
+        destination_cursor.execute("SELECT id FROM public.courts WHERE name = 'Default Court'")
+        default_court_id = destination_cursor.fetchone()
+
+        temp_cases_data = []
+        for case in source_data:
+            reference = case[1]
+            
+            if reference is None:
+                self.failed_imports.add(('cases', case[0]))
+                log_failed_imports(self.failed_imports)
```

Comment:
'Createdby' field is missing.

Reply:
The created_by is recorded in the audits table